Tests/ClientRuntimeTests/ClientRuntimeTests/NetworkingTests/Http/MiddlewareTests/AuthSchemeMiddlewareTests.swift

```diff
@@ -0,0 +1,140 @@
+            let expected = "Could not resolve auth scheme for the operation call. Log: Auth scheme fillerAuth was not enabled for this request."
+            XCTAssertEqual(message, expected)
+        } catch {
+            XCTFail("Unexpected error thrown: \(error.localizedDescription)")
+        }
+    }
+
+    // Test success cases
+    func testOnlyAuthSchemeA() async throws {
+        let context = contextBuilder
+            .withOperation(value: "authA")
+            .withAuthScheme(value: MockAuthSchemeA())
+            .build()
+        try await AssertSelectedAuthSchemeMatches(builtContext: context, expectedAuthScheme: "MockAuthSchemeA")
+    }
+
+    func testAuthOrderABSelectA() async throws {
+        let context = contextBuilder
+            .withOperation(value: "authAB")
+            .withAuthScheme(value: MockAuthSchemeA())
+            .withAuthScheme(value: MockAuthSchemeB())
+            .build()
+        try await AssertSelectedAuthSchemeMatches(builtContext: context, expectedAuthScheme: "MockAuthSchemeA")
+    }
+
+    func testAuthOrderABSelectB() async throws {
+        let context = contextBuilder
+            .withOperation(value: "authAB")
+            .withAuthScheme(value: MockAuthSchemeB())
+            .build()
+        try await AssertSelectedAuthSchemeMatches(builtContext: context, expectedAuthScheme: "MockAuthSchemeB")
+    }
+
+    func testAuthOrderABCSelectA() async throws {
+        let context = contextBuilder
+            .withOperation(value: "authABC")
+            .withAuthScheme(value: MockAuthSchemeA())
+            .withAuthScheme(value: MockAuthSchemeB())
+            .withAuthScheme(value: MockAuthSchemeC())
+            .build()
+        try await AssertSelectedAuthSchemeMatches(builtContext: context, expectedAuthScheme: "MockAuthSchemeA")
+    }
+
+    func testAuthOrderABCSelectB() async throws {
+        let context = contextBuilder
+            .withOperation(value: "authABC")
+            .withAuthScheme(value: MockAuthSchemeB())
+            .withAuthScheme(value: MockAuthSchemeC())
+            .build()
+        try await AssertSelectedAuthSchemeMatches(builtContext: context, expectedAuthScheme: "MockAuthSchemeB")
+    }
+
+    func testAuthOrderABCSelectC() async throws {
+        let context = contextBuilder
+            .withOperation(value: "authABC")
+            .withAuthScheme(value: MockAuthSchemeC())
+            .build()
+        try await AssertSelectedAuthSchemeMatches(builtContext: context, expectedAuthScheme: "MockAuthSchemeC")
+    }
+
+    func testAuthOderABCNoAuthSelectNoAuth() async throws {
+        let context = contextBuilder
+            .withOperation(value: "authABCNoAuth")
+            .build()
+        try await AssertSelectedAuthSchemeMatches(builtContext: context, expectedAuthScheme: "smithy.api#noAuth")
+    }
+
+    private func AssertSelectedAuthSchemeMatches(builtContext: HttpContext, expectedAuthScheme: String) async throws {
```

Comment:
When writing a custom test assertion, capture filename & line number then pass them to the XCTest assertion so that the IDE marks up the failed tests appropriately.

Example: https://github.com/smithy-lang/smithy-swift/blob/main/Tests/ClientRuntimeTests/NetworkingTests/ContentLengthMiddlewareTests.swift#L79

Reply:
Changed as suggested.